components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/IdentityProviderDataRetrievalClient.java

```diff
@@ -102,6 +112,131 @@ public String getIdPImage(String tenant, String idpName)
+     * @throws IdentityProviderDataRetrievalClientException When there is an error with executing the get calls or
+     *                                                      processing JSON objects.
+     */
+    public Map<String, String> getFederatedIdpConfigs(String tenant, String idpCode, String idpName,
+                                                      List<String> configKeys)
+            throws IdentityProviderDataRetrievalClientException {
+
+        Map<String, String> configMap = new HashMap<>();
+        if (StringUtils.isEmpty(tenant) || StringUtils.isEmpty(idpCode) || StringUtils.isEmpty(idpName) ||
+                CollectionUtils.isEmpty(configKeys)) {
+            return configMap;
+        }
+        // i.e. /t/carbon.super/api/server/v1/identity-providers?filter=name+eq+Google
+        JSONObject idpSummaryResult = executePath(tenant, IDP_API_RELATIVE_PATH + IDP_FILTER +
+                Encode.forUriComponent(idpName));
+        if (idpSummaryResult == null) {
+            return configMap;
+        }
+
+        try {
+            JSONArray idpSummary = idpSummaryResult.getJSONArray(IDP_KEY);
+            if (idpSummary.length() != 1) {
+                return configMap;
+            }
+            JSONObject idpSummaryProperties = (JSONObject) idpSummary.get(0);
+            if (idpSummaryProperties == null) {
+                return configMap;
+            }
+            // i.e. /t/carbon.super/api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc
+            String idpURLWithId = idpSummaryProperties.getString(SELF_LINK);
+            if (StringUtils.isEmpty(idpURLWithId)) {
+                return configMap;
+            }
+            JSONObject idpDetailedResult = executePath(tenant, idpURLWithId);
+            if (idpDetailedResult == null) {
+                return configMap;
+            }
+            JSONObject federatedAuthObject = idpDetailedResult.getJSONObject(FEDERATED_AUTHENTICATORS);
+            if (federatedAuthObject == null) {
+                return configMap;
+            }
+            JSONArray federatedAuthenticators = federatedAuthObject.getJSONArray(AUTHENTICATORS);
+            if (federatedAuthenticators == null) {
+                return configMap;
+            }
+            String federatedIDPURLWithID = null;
+            for (int i = 0; i < federatedAuthenticators.length(); i++) {
+                JSONObject property = (JSONObject) federatedAuthenticators.get(i);
+                if (idpCode.equals(property.getString(NAME))) {
+                    federatedIDPURLWithID = property.getString(SELF_LINK);
+                    break;
+                }
+            }
+            if (StringUtils.isEmpty(federatedIDPURLWithID)) {
+                return configMap;
+            }
+
+            // i.e. /t/carbon.super/api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc/
+            // federated-authenticators/R29vZ2xlT0lEQ0F1dGhlbnRpY2F0b3I
```

Comment:
Shall we use a block comment here?

Reply:
Fixed 38d0c51fd441a7ee513b0b52d2459c3a732d9e2d